### PR TITLE
Trey tweaks update for nise cdr cetb

### PR DESCRIPTION
**Pull request by trey-stafford**
* Fixed up some ignored type errors
* Move some a2l1c related functions out of `bt.cli` and into separate module. I think this will help with the transition to using `pm_tb_data`.

### CODE REVIEW

File: pm_icecon/bt/params/amsr2.py
Comment (trey-stafford) [inline pm_icecon/bt/params/amsr2.py:1]: Lets talk about this and the `amsr2_goddard.py`. It's not clear to me where the differences came from and when each should be used. They both indicate that they originally came from `ret_parameters_amsru2.f`

File: pm_icecon/_types.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
-from typing import Literal, Union
+from typing import Literal
 
-Hemisphere = Union[Literal['north'], Literal['south'], str]
+Hemisphere = Literal['north', 'south']
```
Comment (sc0tts) [inline pm_icecon/_types.py:3]: Hmmm.  I had needed to add the `str` condition when calling Hemisphere using a variable in a function.  I wonder if that still works when the `str` is removed?

File: pm_icecon/bt/params/amsr2.py
Comment (sc0tts) [reply pm_icecon/bt/params/amsr2.py:1]: Yes, let's chat.  Everything is identical except some of the weather filter coefficients.  We did some runs and found that we did a better job of removing observed weather in the polar_stereo 12.5km AMSR2 sea ice concentrations calculated from the AU_SI12 data when we used our coefficients than the ones Goddard provided.

File: pm_icecon/bt/params/amsr2.py [hunk 2]
```diff
@@ -35,7 +36,9 @@
             start_month=11,
             end_month=4,
             weather_filter_params=WeatherFilterParams(
-                wintrc=84.73, wslope=0.5352, wxlimt=13.7
+                wintrc=84.73,
+                wslope=0.5352,
+                wxlimt=13.7,
             ),
```
Comment (sc0tts) [inline pm_icecon/bt/params/amsr2.py:42]: Lol.  I tried to have these on separate lines, but one of the linters -- I thought it was black? -- put them on the same line....

**Review by sc0tts — APPROVED** [review]
Assuming all the tests work, these changes look good to me.

File: pm_icecon/bt/params/amsr2.py (hunk 2 above)
Comment (trey-stafford) [reply pm_icecon/bt/params/amsr2.py:42]: Adding a trailing `,` to the last kwarg tells black to format this way. Not really a big deal, but I like having each kwarg on it's on line :)

File: pm_icecon/_types.py (hunk 1 above)
Comment (trey-stafford) [reply pm_icecon/_types.py:3]: All of the tests are passing so I think my changes resolved the issue. The way we are handling hemispheres should be changing very soon though.

**Pull request merged**